packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +100,123 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// Represents a range of content for a given [SelectedContent].
+///
+/// This class stores the information of an active selection under
+/// a [Selectable] or [SelectionHandler], including the start and
+/// end positions of a selection relative to the content, as well
+/// as an identifier to identify the owner of the content, any
+/// children selection ranges that may be contained within the active
+/// selection, and the content itself.
+///
+/// [SelectionArea] and [SelectableRegion] provide access to this
+/// information through the [SelectedContent] provided in their
+/// [SelectionArea.onSelectionChanged] and
+/// [SelectableRegion.onSelectionChanged] members.
+///
+/// {@tool dartpad}
+/// This example shows how to color red the active selection
+/// under a [SelectionArea] or [SelectableRegion].
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.1.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to replace the active selection
+/// under a [SelectionArea] or [SelectableRegion] with a widget.
+///
+/// ** See code in examples/api/lib/material/selection_area/selection_area.2.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [SelectedContent], which contains the [SelectedContentRange] for a given [Selectable] or [SelectionHandler].
+abstract class SelectedContentRange<T extends Object> {
+  /// Creates a range for the content of a [Selectable] or [SelectionHandler].
+  SelectedContentRange({
+    this.selectableId,
+    required this.content,
+    this.children,
+  });
+
+  /// The unique id for the [Selectable] that created the range.
+  final int? selectableId;
+
+  /// The content that contains the selection.
+  final T content;
+
+  /// The start of the selection relative to the [content].
+  int get startOffset;
+
+  /// The end of the selection relative to the [content].
+  int get endOffset;
+
+  /// Additional ranges to include as children.
+  ///
+  /// Children of a given range enable more granular modification of the
+  /// selection.
+  final List<SelectedContentRange<Object>>? children;
+
+  @override
+  String toString() {
+    return 'SelectedContentRange(\n'
+           '  selectableId: $selectableId,\n'
+           '  content: $content,\n'
+           '  startOffset: $startOffset,\n'
+           '  endOffset: $endOffset,\n'
+           '  children: $children,\n'
+           ')';
+  }
+}
+
 /// The selected content in a [Selectable] or [SelectionHandler].
 // TODO(chunhtai): Add more support for rich content.
 // https://github.com/flutter/flutter/issues/104206.
 class SelectedContent {
   /// Creates a selected content object.
   ///
   /// Only supports plain text.
-  const SelectedContent({required this.plainText});
+  const SelectedContent({
```

Comment:
SelectedContent and SelectedContentRange seem very similar. We should be very clear about their roles in relation to one another.

Is the fundamental reason they are separate classes because SelectedContent contains layout information (`geometry`) and SelectedContentRange does not?

Reply:
Now this SelectedContent class is a lot simpler and better IMO. But this PR is still making the change that this goes from wrapping text to wrapping text and associated layout information.

@LongCatIsLooong I wonder if you can think of any reason not to do that? @Renzo-Olivares could a SelectedContent instance outlive the frame in which the `geometry` is relevant in a way that would be hard to reconcile?

Reply:
The geometry is being used in the examples to show a custom context menu at the selection.

Reply:
Now geometry is gone and this is back to how it was before this PR, looks good :+1: